lib/srv/sess.go

```diff
@@ -1327,18 +1324,6 @@ func (m *multiWriter) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
-func (m *multiWriter) Close() error {
```

Comment:
This might deadlock in session `Close` as session also registers a `party` as `io.Closer` and multiWriter gets to lock the mutex twice.  Does not seem to be necessary though as the resource added as writers here are also cleaned up elsewhere. Although it's a bit of entanglement so any advice or warning here is welcome.

I'm working on moving the clean up of `s.recorder` explicitly to `session.Close`.

Reply:
Where are these `m.writers` closed?

Reply:
> Where are these `m.writers` closed?

There were just two: [session recorder](https://github.com/gravitational/teleport/blob/dmitri/5970-auditlog-race/lib/srv/sess.go#L695) which I moved to `session.Close` to dispose of explicitly and another one is the party [here](https://github.com/gravitational/teleport/blob/dmitri/5970-auditlog-race/lib/srv/sess.go#L1227) which is also registered for clean up in the context [here](https://github.com/gravitational/teleport/blob/dmitri/5970-auditlog-race/lib/srv/sess.go#L1228) so it seems redundant.